### PR TITLE
[EraVM][AsmParser] Cleanup outdated mnemonic aliases

### PR DESCRIPTION
**Pull request by atrosinenko**
Remove `MnemonicAlias`es as they are not needed in the new syntax and are only used on asm parsing.

### CODE REVIEW

File: llvm/lib/Target/EraVM/AsmParser/EraVMAsmParser.cpp [hunk 1]
```diff
@@ -289,9 +289,6 @@ bool EraVMAsmParser::MatchAndEmitInstruction(SMLoc Loc, unsigned &Opcode,
 // Auto-generated by TableGen
 static unsigned MatchRegisterName(StringRef Name);
 static unsigned MatchRegisterAltName(StringRef Name);
-static void applyMnemonicAliases(StringRef &Mnemonic,
```
Comment (asl) [inline llvm/lib/Target/EraVM/AsmParser/EraVMAsmParser.cpp:292]: Maybe keep the declaration and call? So if we'd add some aliases in the future they will apply automatically?

Comment (atrosinenko) [reply llvm/lib/Target/EraVM/AsmParser/EraVMAsmParser.cpp:292]: Unfortunately, it seems `applyMnemonicAliases` is only generated if at least one alias exists. Maybe defining a dummy weak function would help, but I doubt it works in the same translation unit.

Comment (asl) [reply llvm/lib/Target/EraVM/AsmParser/EraVMAsmParser.cpp:292]: Ah, ok, makes sense. Ok, then!

**Review by sayon — APPROVED** [review]
LGTM, thank you!

**Pull request merged**